api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -0,0 +1,437 @@
+import logging
+import os
+import sys
+from datetime import timedelta
+from time import time
+from typing import Any, Dict, Iterator, Optional, Sequence, Tuple
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+import scipy
+import somacore
+import tiledbsoma as soma
+import torch
+import torchdata.datapipes.iter as pipes
+from attr import attrs
+from scipy import sparse
+from sklearn.preprocessing import LabelEncoder
+from somacore.query import _fast_csr
+from torch import Tensor
+from torch.utils.data import DataLoader
+from torch.utils.data.dataset import Dataset
+
+import cellxgene_census
+from cellxgene_census._open import _build_soma_tiledb_context
+
+ObsDatum = Tuple[Tensor, torch.sparse_coo_tensor]
+
+Encoders = Dict[str, LabelEncoder]
+
+DEFAULT_BUFFER_BYTES = 1024**3
+
+DEFAULT_WORKER_TIMEOUT = 120
+
+pytorch_logger = logging.getLogger("cellxgene_census.experimental.pytorch")
```

Comment:
I like that you are using the logger.  Should we expand this to other parts of the code base?

Reply:
suggest a new issue for future debt work.  The builder also has lots of sloppy use of logger (my fault!)

Reply:
https://github.com/chanzuckerberg/cellxgene-census/issues/504